source/concurrent/Promise.ooc

```diff
@@ -109,7 +109,7 @@ Future: abstract class <T> {
 }
 
 _ThreadFuture: class <T> extends Future<T> {
-	_result: Object
+	_result: __onheap__ T
```

Comment:
This is actually exactly the keyword that `Cell` uses to make sure its contents doesn't disappear in a long lost stack frame.
We should really change the keyword to something prettier, though.